lib/dao.js

```diff
@@ -757,12 +757,12 @@ DataAccessObject.findOrCreate = function findOrCreate(query, data, options, cb)
 
         // options has precedence on model-setting
         if (options.validate === false) {
-          _findOrCreate(query, data, obj);
+          return _findOrCreate(query, data, obj);
```

Comment:
In fact if we don't have this fix, 16 test cases fail; for instance, [this](https://github.com/strongloop/loopback-datasource-juggler/blob/master/test/optional-validation.test.js#L435-L437) is one of the test cases which fails; in other words, as long as we have the implementation for 'findOrCreate' in the connector and the code does not run the [workaround](https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/dao.js#L779-L797) we have these failures; so I believe test cases cover this. Please let me know if you think otherwise?

PS: I commit this fix separately...